### ISSUE TITLE
abstract def compilation errors with Crystal 0.36.0

### ISSUE THREAD

**Issue by matthewmcgarvey** [amberframework/granite#440]
Granite: 0.21.1

After updating to Crystal 0.36.0 I started getting compilation errors. I believe it is because the abstract defs do not declare type restrictions on their arguments while the implementations do.

Here's an example:

`Granite::Adapter::Base#clear` https://github.com/amberframework/granite/blob/af10c182d374e62df82af83a3a55f5fbdba990a7/src/adapter/base.cr#L36
`Granite::Adapter::Pg#clear` https://github.com/amberframework/granite/blob/af10c182d374e62df82af83a3a55f5fbdba990a7/src/adapter/pg.cr#L30

Error:

```bash
Showing last frame. Use --error-trace for full trace.

In lib/granite/src/adapter/pg.cr:5:1

 5 | class Granite::Adapter::Pg < Granite::Adapter::Base
     ^
Error: abstract `def Granite::Adapter::Base#clear(table_name)` must be implemented by Granite::Adapter::Pg
```

**Comment by matthewmcgarvey** [comment]
https://github.com/amberframework/granite/pull/439 has the fixes for this issue but has failing specs

**Comment by drujensen** [comment]
Released 0.22.1 that supports Crystal 0.36.x
